### PR TITLE
upkeep:  Add TS support for `PostTitle` component

### PR DESCRIPTION
**Pull request by Sidsector9**
### Description of the Change

Closes #

### Screenshot

Marks invalid attributes:

Screenshot 2024-03-04 at 11 29 40 AM

### Changelog Entry

> Changed - Add TS support for `PostTitle` component.

### Credits

Props @fabiankaegy @Sidsector9 

### Checklist:

- [ ] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: components/post-title/index.tsx [hunk 1]
```diff
@@ -1,11 +1,18 @@
...
+interface PostTitleProps<T extends React.ElementType> {
+	tagName?: T;
+}
+
+type PostTitlePropsWithOmit<T extends React.ElementType> = PostTitleProps<T> & Omit<React.ComponentPropsWithoutRef<T>, keyof PostTitleProps<T>>;
+
+export const PostTitle = <T extends React.ElementType = 'h1'>({
+	tagName,
+	...rest
+}: PostTitlePropsWithOmit<T> ) => {
```
Comment (fabiankaegy) [inline components/post-title/index.tsx]: Not at all critical, and I am very open to being wrong here. My personal preference over single letter type generics is having a more descriptive name. 
```suggestion
interface PostTitleProps<T extends React.ElementType> {
	tagName?: T;
}

type PostTitlePropsWithOmit<TElementType extends React.ElementType> = PostTitleProps<TElementType> & Omit<React.ComponentPropsWithoutRef<TElementType>, keyof PostTitleProps<TElementType>>;

export const PostTitle = <TElementType extends React.ElementType = 'h1'>({
	tagName,
	...rest
}: PostTitlePropsWithOmit<TElementType> ) => {
```

File: components/post-title/index.tsx [hunk 2]
```diff
@@ -1,11 +1,18 @@
...
-export const PostTitle = (props) => {
-	const { tagName: TagName = 'h1', ...rest } = props;
+interface PostTitleProps<T extends React.ElementType> {
+	tagName?: T;
+}
+
+type PostTitlePropsWithOmit<T extends React.ElementType> = PostTitleProps<T> & Omit<React.ComponentPropsWithoutRef<T>, keyof PostTitleProps<T>>;
+
+export const PostTitle = <T extends React.ElementType = 'h1'>({
+	tagName,
```
Comment (fabiankaegy) [inline components/post-title/index.tsx]: I would move the default value away from PropTypes into the default inline props:

```suggestion
	tagName = 'h1',
```

File: components/post-title/index.tsx [hunk 3]
```diff
@@ -37,10 +46,6 @@ export const PostTitle = (props) => {
 	);
 };
 
-PostTitle.propTypes = {
-	tagName: PropTypes.string,
-};
-
 PostTitle.defaultProps = {
 	tagName: 'h1',
```
Comment (fabiankaegy) [inline components/post-title/index.tsx]: Lets move this to the inline assignment (default value) so we can remove prop types completely

Comment (Sidsector9) [reply components/post-title/index.tsx]: @fabiankaegy this PR isn't ready yet, I'm having a hard time assigning that as a default value.

I get this error:

```
Type '"h1"' is not assignable to type 'TElementType'.
  '"h1"' is assignable to the constraint of type 'TElementType', but 'TElementType' could be instantiated with a different subtype of constraint 'ElementType<any>'.ts(2322)
```

which makes sense but unsure on how to resolve it. I'll mark this PR as draft.

File: components/post-title/index.tsx [hunk 4]
```diff
@@ -37,10 +46,6 @@ export const PostTitle = (props) => {
...
 };
```
Comment (fabiankaegy) [inline components/post-title/index.tsx]: @Sidsector9 Should we not remove the Default Props here?

Comment (Sidsector9) [reply components/post-title/index.tsx]: Hey thanks for taking care of this one, I was on PTO.

**Pull request merged**